ios/FluentUI/AvatarGroup/AvatarGroup.swift

```diff
@@ -112,78 +113,90 @@ public struct AvatarGroup: View, ConfigurableTokenizedControl {
-            let overflowTokens = overflowAvatar.tokens
-            let overflowMaxHeight = overflowTokens.avatarSize + 2 * (overflowTokens.ringThickness + overflowTokens.ringInnerGap + overflowTokens.ringOuterGap)
+            let avatarSize = overflowAvatar.tokenSet[.avatarSize].float
+            let ringThickness = overflowAvatar.tokenSet[.ringThickness].float
+            let ringInnerGap = overflowAvatar.tokenSet[.ringInnerGap].float
+            let ringOuterGap = overflowAvatar.tokenSet[.ringOuterGap].float
+
+            let overflowMaxHeight = avatarSize + (2 * ringThickness) + ringInnerGap + ringOuterGap
 
             return max(avatarMaxHeight, overflowMaxHeight)
         }()
 
+        @ViewBuilder
+        func avatarViewInHStack(_ index: Int, _ avatar: MSFAvatarStateImpl) -> some View {
+            let nextIndex = index + 1
+            let isLastDisplayed = nextIndex == avatarsToDisplay
+            // If the avatar is part of Stack style and is not the last avatar in the sequence, create a cutout.
+            let avatarView = avatarViews[index]
+            let needsCutout = isStackStyle && (hasOverflow || !isLastDisplayed)
+            let avatarSize: CGFloat = avatarView.totalSize
+            let nextAvatarSize: CGFloat = isLastDisplayed ? overflowAvatar.totalSize : avatarViews[nextIndex].totalSize
+
+            // Calculating the size delta of the current and next avatar based off of ring visibility, which helps determine
+            // starting coordinates for the cutout.
+            let currentAvatarHasRing = avatar.isRingVisible
+            let nextAvatarHasRing = !isLastDisplayed ? avatars[nextIndex].isRingVisible : false
+
+            // Calculating the different interspace scenarios considering rings.
+            let ringOuterGap = avatarView.tokenSet[.ringOuterGap].float
+            let ringOffset = avatarView.tokenSet[.ringInnerGap].float + avatarView.tokenSet[.ringThickness].float + ringOuterGap
+            let stackPadding = interspace - (currentAvatarHasRing ? ringOffset : 0) - (nextAvatarHasRing ? ringOuterGap : 0)
+
+            // Finalized calculations for x and y coordinates of the Avatar if it needs a cutout, including RTL.
+            let ringGapOffset = 2 * ringOuterGap
+            let xOrigin: CGFloat = {
+                if layoutDirection == .rightToLeft {
+                    return -nextAvatarSize - interspace + ringOuterGap + (currentAvatarHasRing ? ringOffset : 0)
+                }
+                return avatarSize + interspace - ringGapOffset - ringOuterGap - (currentAvatarHasRing ? ringOuterGap : 0)
+            }()
+
+            let sizeDiff = avatarSize - nextAvatarSize - (currentAvatarHasRing ? 0 : ringGapOffset)
+            let yOrigin = sizeDiff / 2
+
+            // Hand the rendering of the avatar to a helper function to appease Swift's
+            // strict type-checking timeout.
+            VStack {
```

Comment:
does "avatarViewInHStack" naming make sense if we have vstack stuff going on inside this block?

Reply:
Fair, not the best name. I'll get a better one.

Reply:
New signature:
```swift
func avatarView(at index: Int, for avatar: MSFAvatarStateImpl)
```